elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java

```diff
@@ -5,26 +5,172 @@
  */
 package com.yahoo.elide.core.filter;
 
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.PersistentResource;
 import com.yahoo.elide.core.exceptions.InvalidPredicateException;
+import com.yahoo.elide.utils.coerce.CoerceUtil;
+
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
 /**
  * Operator enum for predicates.
  */
 @RequiredArgsConstructor
-public enum Operator {
-    IN("in", true),
-    NOT("not", true),
-    PREFIX("prefix", true),
-    POSTFIX("postfix", true),
-    INFIX("infix", true),
-    ISNULL("isnull", false),
-    NOTNULL("notnull", false),
-    LT("lt", true),
-    LE("le", true),
-    GT("gt", true),
-    GE("ge", true);
+public enum Operator implements BiFunction<Predicate, EntityDictionary, java.util.function.Predicate> {
```

Comment:
I totally agree with you. I like the idea of changing elide's Predicate to FilterPredicate. Having conflicting names with java builtin packages is such a pain. Do you think it would be a breaking change to rename? If it's not I'm glad to change